internal/connector/pingone/sso/resources/pingone_resource_secret.go

```diff
@@ -0,0 +1,118 @@
+package resources
+
+import (
+	"fmt"
+
+	"github.com/patrickcping/pingone-go-sdk-v2/management"
+	"github.com/pingidentity/pingcli/internal/connector"
+	"github.com/pingidentity/pingcli/internal/connector/common"
+	"github.com/pingidentity/pingcli/internal/connector/pingone"
+	"github.com/pingidentity/pingcli/internal/logger"
+)
+
+// Verify that the resource satisfies the exportable resource interface
+var (
+	_ connector.ExportableResource = &PingOneResourceSecretResource{}
+)
+
+type PingOneResourceSecretResource struct {
+	clientInfo *connector.PingOneClientInfo
+}
+
+// Utility method for creating a PingOneResourceSecretResource
+func ResourceSecret(clientInfo *connector.PingOneClientInfo) *PingOneResourceSecretResource {
+	return &PingOneResourceSecretResource{
+		clientInfo: clientInfo,
+	}
+}
+
+func (r *PingOneResourceSecretResource) ResourceType() string {
+	return "pingone_resource_secret"
+}
+
+func (r *PingOneResourceSecretResource) ExportAll() (*[]connector.ImportBlock, error) {
+	l := logger.Get()
+	l.Debug().Msgf("Exporting all '%s' Resources...", r.ResourceType())
+
+	importBlocks := []connector.ImportBlock{}
+
+	resourceData, err := r.getResourceData()
+	if err != nil {
+		return nil, err
+	}
+
+	for resourceId, resourceName := range resourceData {
+		resourceSecretOk, err := r.getResourceSecret(resourceId)
+		if err != nil {
+			return nil, err
+		}
+
+		if !resourceSecretOk {
+			continue
+		}
+
+		commentData := map[string]string{
+			"Resource ID":           resourceId,
+			"Resource Name":         resourceName,
+			"Export Environment ID": r.clientInfo.ExportEnvironmentID,
+			"Resource Type":         r.ResourceType(),
+		}
+
+		importBlock := connector.ImportBlock{
+			ResourceType:       r.ResourceType(),
+			ResourceName:       fmt.Sprintf("%s_secret", resourceName),
+			ResourceID:         fmt.Sprintf("%s/%s", r.clientInfo.ExportEnvironmentID, resourceId),
+			CommentInformation: common.GenerateCommentInformation(commentData),
+		}
+
+		importBlocks = append(importBlocks, importBlock)
+	}
+
+	return &importBlocks, nil
+}
+
+func (r *PingOneResourceSecretResource) getResourceData() (map[string]string, error) {
+	resourceData := make(map[string]string)
+
+	iter := r.clientInfo.ApiClient.ManagementAPIClient.ResourcesApi.ReadAllResources(r.clientInfo.Context, r.clientInfo.ExportEnvironmentID).Execute()
+	resourceInners, err := pingone.GetManagementAPIObjectsFromIterator[management.EntityArrayEmbeddedResourcesInner](iter, "ReadAllResources", "GetResources", r.ResourceType())
+	if err != nil {
+		return nil, err
+	}
+
+	for _, resourceInner := range resourceInners {
+		if resourceInner.Resource != nil {
+			resourceId, resourceIdOk := resourceInner.Resource.GetIdOk()
+			resourceName, resourceNameOk := resourceInner.Resource.GetNameOk()
+			resourceType, resourceTypeOk := resourceInner.Resource.GetTypeOk()
+
+			if resourceIdOk && resourceNameOk && resourceTypeOk && *resourceType == management.ENUMRESOURCETYPE_CUSTOM {
```

Comment:
Is the `resource_secret` resource limited to `CUSTOM` resource types only. I couldn't find any info saying so in a quick look at the registry docs: https://registry.terraform.io/providers/pingidentity/pingone/latest/docs/resources/resource_secret